qa/entitlements/src/javaRestTest/java/org/elasticsearch/test/entitlements/EntitlementsIT.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.test.entitlements;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.containsString;
+
+@ESTestCase.WithoutSecurityManager
+public class EntitlementsIT extends ESRestTestCase {
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.INTEG_TEST)
+        .plugin("entitlement-qa")
+        .systemProperty("es.entitlements.enabled", "true")
+        .setting("xpack.security.enabled", "false")
+        .jvmArg("-Dentitlement.test=true")
+        .jvmArg("-Djdk.attach.allowAttachSelf=true")
+        .jvmArg("-XX:+EnableDynamicAgentLoading")
+        .jvmArg("--patch-module=java.base=lib/entitlement-bridge/elasticsearch-entitlement-bridge-9.0.0-SNAPSHOT.jar")
```

Comment:
Should this version and `SNAPSHOT` thing be computed rather than hardcoded?

Reply:
Yeah, I copy/pasted what I needed to make this work :D 
I'll copy/reuse the correct logic.

Reply:
I did not reused the same logic in the main code, situation here is a bit different. I used gradle to get the name of the compiled jar; I'll ask Mark to review it.